sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java

```diff
@@ -32,16 +33,23 @@
 @ServiceClient(builder = OpenAIClientBuilder.class, isAsync = true)
 public final class OpenAIAsyncClient {
 
-    @Generated private final OpenAIClientImpl serviceClient;
+    private final OpenAIClientImpl serviceClient;
+
+    private final OpenAIClientNonAzureImpl openAIServiceClient;
 
     /**
      * Initializes an instance of OpenAIAsyncClient class.
      *
      * @param serviceClient the service client implementation.
      */
-    @Generated
     OpenAIAsyncClient(OpenAIClientImpl serviceClient) {
         this.serviceClient = serviceClient;
+        this.openAIServiceClient = null;
+    }
+
+    OpenAIAsyncClient(OpenAIClientNonAzureImpl openAIServiceClient) {
```

Comment:
Missing javadoc. I would add a quick word on why we set one `serviceClient` to null in the case of a non-Azure OpenAI credential setup, just to be very clear that the `OpenAIClient` operates either way in a mutually exclusive way.